dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/common/info/OpenTelemetryInfoFactoryImpl.java

```diff
@@ -76,23 +73,37 @@ public class OpenTelemetryInfoFactoryImpl implements ApplicationStateListener, O
     private static final TraceComponent tc = Tr.register(OpenTelemetryInfoFactoryImpl.class);
 
     private final MetaDataSlot slotForOpenTelemetryInfoHolder;
+    Map<String, OpenTelemetry> otelMap = new HashMap<String, OpenTelemetry>();
 
     //This contains API calls that change between the upstream open telemetry version.
     //We get a partially configued SDK Builder from OSGi becase we are in a static context
     //and do not know which version of mpTelemetry will be in use.
-    @Reference
-    private OpenTelemetryVersionedConfiguration openTelemetryVersionedConfiguration;
+    private final OpenTelemetryVersionedConfiguration openTelemetryVersionedConfiguration;
 
     @Activate
-    public OpenTelemetryInfoFactoryImpl(@Reference MetaDataSlotService slotService) {
+    public OpenTelemetryInfoFactoryImpl(@Reference MetaDataSlotService slotService,
+                                        @Reference OpenTelemetryVersionedConfiguration openTelemetryVersionedConfiguration) {
+
         slotForOpenTelemetryInfoHolder = slotService.reserveMetaDataSlot(ApplicationMetaData.class);
+
+        this.openTelemetryVersionedConfiguration = openTelemetryVersionedConfiguration;
+
+        OpenTelemetry runtimeInstance = this.openTelemetryVersionedConfiguration.createServerOpenTelemetryInfo(getServerTelemetryProperties());
+
+        if (runtimeInstance != null)
+            otelMap.put(OpenTelemetryConstants.OTEL_RUNTIME_INSTANCE_NAME, runtimeInstance);
+        else
+            otelMap.put(OpenTelemetryConstants.OTEL_RUNTIME_INSTANCE_NAME, null);
     }
 
     @Override
     public OpenTelemetryInfo getOpenTelemetryInfo() {
         try {
-            ApplicationMetaData metaData = ComponentMetaDataAccessorImpl.getComponentMetaDataAccessor().getComponentMetaData().getModuleMetaData().getApplicationMetaData();
-
+            ApplicationMetaData metaData = null;
+            ComponentMetaData cmd = ComponentMetaDataAccessorImpl.getComponentMetaDataAccessor().getComponentMetaData();
+            if (cmd != null) {
+                metaData = cmd.getModuleMetaData().getApplicationMetaData();
+            }
             return getOpenTelemetryInfo(metaData);
```

Comment:
`metaData` could be `null`.  Does `getOpenTelemetryInfo` handle `null`?

Reply:
Yeah, this is the original OTEL code. If metadata is null, it returns an exception